ivadomed/utils.py

```diff
@@ -834,7 +834,7 @@ def __call__(self, sample):
                 return False
 
         if self.filter_empty_input:
-            if not np.all([np.any(img) for img in input_data]):
+            if not np.all([np.any(img) for img in input_data]) or not np.all([img.std() != 0 for img in input_data]):
```

Comment:
The double negation is a bit confusing, without negations I find it clearer.

```suggestion
            if np.any([np.all(img) for img in input_data]) or np.any([img.std() == 0 for img in input_data]):
```

Reply:
Actually... `np.any([img.std() == 0 for img in input_data])` covers `np.any([np.all(img) for img in input_data])` eheh we can make it even simpler I think.

Reply:
Yes you're absolutely right!